Setup of mxd tutorial (#96)
eclipse-tractusx/tutorial-resources: mxd/2_MXD_Setup.md

# Basic dataspace setup

The "MXD" dataspace initially consists of several components: `Alice` and `Bob` (two Tractus-X EDC connectors),

> a `vault` instance each, a Postgres database, a Managed Identity Wallet app, a Keycloak instance. `Alice` and `Bob` will

Reviewer: ```suggestion
a Vault instance each, a Postgres database, a Managed Identity Wallet app, a Keycloak instance. `Alice` and `Bob` will
```

> be our dataspace participants. Each of them stores their secrets in their "private" vault instance, and there is a

Reviewer: ```suggestion
be our dataspace participants. Each of them stores their secrets in their respective vault instances, and there is a
```

> shared Postgres server, where each of them has a database. MIW and Keycloak are "central" components, they only exist

Reviewer: ```suggestion
shared Postgres server, where each of them has a database. MIW and Keycloak are central components, they only exist
```

kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
# wait until the ingress controller is ready
kubectl wait --namespace ingress-nginx \
  --for=condition=ready pod \
  --selector=app.kubernetes.io/component=controller \
  --timeout=90s
terraform init
terraform apply
# type "yes" and press enter when prompted to do so 
```

Notice that the `kubectl apply` command deploys a Kubernetes Ingress Controller to the cluster and is required to reach
our applications from outside the cluster. Specifically, it deploys an NGINX ingress controller. Notice also, that the
command is *specific to KinD* and will likely not work on other Kubernetes runtimes (minikube, ...) or with other
ingress controllers!

Wait. Then wait some more. It will take a couple of minutes until all services are booted up. If your machine is a
potato, it'll take even longer. Just get a coffee. Eventually, it should look similar to this:

![img.png](assets/img.png)

## Inspect terraform output

After the `terraform` command has successfully completed, it will output a few configuration and setup values

> that we will need in later steps. Please note that most values will be different on your local system.

Reviewer: ```suggestion
that we will need in later steps. Please note that some values will be different on your local system.
```
